anime-list-master.xml

```diff
@@ -13749,11 +13754,11 @@
   <anime anidbid="4728" tvdbid="76703" defaulttvdbseason="10" episodeoffset="" tmdbid="" imdbid="">
     <name>Pocket Monsters: Diamond and Pearl</name>
     <mapping-list>
-      <mapping anidbseason="0" tvdbseason="0">;1-0;2-14;3-20;4-0;5-0;</mapping>
-      <mapping anidbseason="1" tvdbseason="10">;1-1;2-2;3-3;4-4;5-5;6-6;7-7;8-8;9-9;10-10;11-11;12-12;13-13;14-14;15-15;16-16;17-17;18-18;19-19;20-20;21-21;22-22;23-23;24-24;25-25;26-26;27-27;28-28;29-29;30-30;31-31;32-32;33-33;34-34;35-35;36-36;37-37;38-38;39-39;40-40;41-41;42-42;43-43;44-44;45-45;46-46;47-47;48-48;49-49;50-50;51-51;52-52;</mapping>
-      <mapping anidbseason="1" tvdbseason="11">;53-1;54-2;55-3;56-4;57-5;58-6;59-7;60-8;61-9;62-10;63-11;64-12;65-13;66-14;67-15;68-16;69-17;70-18;71-19;72-20;73-21;74-22;75-23;76-24;77-25;78-26;79-27;80-28;81-29;82-30;83-31;84-32;85-33;86-34;87-35;88-36;89-37;90-38;91-39;92-40;93-41;94-42;95-43;96-44;97-45;98-46;99-47;100-48;101-49;102-50;103-51;104-52;</mapping>
-      <mapping anidbseason="1" tvdbseason="12">;105-1;106-2;107-3;108-4;109-5;110-6;111-7;112-8;113-9;114-10;115-11;116-12;117-13;118-14;119-15;120-16;121-17;122-18;123-19;124-20;125-21;126-22;127-23;128-24;129-25;130-26;131-27;132-28;133-29;134-30;135-31;136-32;137-33;138-34;139-35;140-36;141-37;142-38;143-39;144-40;145-41;146-42;147-43;148-44;149-45;150-46;151-47;152-48;153-49;154-50;155-51;156-52;157-53;</mapping>
-      <mapping anidbseason="1" tvdbseason="13">;158-1;159-2;160-3;161-4;162-5;163-6;164-7;165-8;166-9;167-10;168-11;169-12;170-13;171-14;172-15;173-16;174-17;175-18;176-19;177-20;178-21;179-22;180-23;181-24;182-25;183-26;184-27;185-28;186-29;187-30;188-31;189-32;190-33;191-34;</mapping>
+      <mapping anidbseason="0" tvdbseason="0">;1-0;2-14;3-20;4-51;5-52;</mapping>
+      <mapping anidbseason="1" tvdbseason="10">;48-0;49-48;50-49;51-50;52-51;</mapping>
+      <mapping anidbseason="1" tvdbseason="11" start="53" end="104" offset="-52" />
+      <mapping anidbseason="1" tvdbseason="12" start="105" end="157" offset="-105">;105-1;106-2;107-3;108-4;109-5;110-6;111-7;112-8;113-9;114-10;115-11;116-12;117-13;118-14;119-15;120-0;</mapping>
```

Comment:
Could this be a separate season mapping instead of the 105..119 individuals?
You did such a great job cleaning up everything else with offsets, it feels almost a shame to keep these many sequential individual mappings
```suggestion
      <mapping anidbseason="1" tvdbseason="12" start="105" end="119" offset="-104">;120-0;</mapping>
      <mapping anidbseason="1" tvdbseason="12" start="121" end="157" offset="-105" />
```

Reply:
I was a bit uncertain about if some less stellar parsers would break if there were multiple mappings for the same anidb & tvdb seasons (can't remember if I've seen it anywhere else), but I'm pretty sure the suggested change at least works with my parser, so I like it :slightly_smiling_face:

Reply:
For some reason the "commit changes" via github didn't work, but I've added a commit with suggested change.

Reply:
This _should_ work for most\all parsers, offsets have been documented and used for a long time, even before this fork.